### ISSUE TITLE
java_image: Escaping $FOO in jvm_args with $ leads to $$FOO in output text

### ISSUE THREAD

**Issue by mtmaxwell** [bazelbuild/rules_docker#853]
I am using the `java_image` rule to containerize pre-existing code that uses environment variables passed via command line arguments.  When I run Bazel on something like the following:

```
	jvm_flags = [
		"-Djvm.variable=$FOO",
        ],
```

I get an error back from Bazel:  _'$FOO' syntax is not supported; use '$(FOO)' instead for "Make" variables, or escape the '$' as '$$' if you intended this for the shell._

(This also happens with `${FOO}` style syntax.)  I tried to escape the `$` as recommended by the error message:

```
	jvm_flags = [
		"-Djvm.variable=$$FOO",
        ],
```
Bazel will now report success.  However, the resulting command-line when run includes `$$FOO`, and not `$FOO` as expected.

**Comment by mtmaxwell** [comment]
I think this may be caused by PR #472

**Comment by nlopezgi** [comment]
Hi @mtmaxwell ,
Thanks for reporting this issue,
iiuc the `$FOO` syntax used to be supported in older versions of Bazel but is no longer supported. My impression is that your use of 
```
      jvm_flags = [
		"-Djvm.variable=$FOO",
        ],
```
does not seem to be any longer supported with new versions of Bazel (i.e., this is not a regression on these rules, it was something that worked 'by accident' before but that now Bazel explicitly disallows, it seems).

Did you try using $(FOO) instead? This should work if make variable substitution is set up for jvm_flags, which it is not at the moment (iiuc), but should not be hard to add (i.e., similar to  #472 which is adding a call to ctx.expand_location, this would also need to call ctx.expand_make_variables). The downside here is that you would need to pass FOO as a --define flag and changing its value would cause a rebuild.

**Comment by ehthayer** [comment]
we looked at $(FOO) but the var FOO isn't known at build time, but is only bound at runtime (is an env var for the container).

We tracked down the behavior that we wanted from the (unfortunately deprecated) ctx.expand_make_variables documentation and forked the repo to make this call instead https://github.com/cargurus/rules_docker/blob/master/java/image.bzl#L215.   Even though this function is deprecated, we weren't able to find a suitable replacement for that functionality it provides.  Also, it wasn't clear to me why ctx.expand_location didn't already include the mapping of  $-escape mapping. Maybe a bug?

The suggestion in the docs for what to replace calls to ctx.expand_make_variables() doesn't seem to address the issue we are having.
